### ISSUE TITLE
Dockerfile does not create working container

### ISSUE THREAD

**Issue by jonathonf** [naturalcrit/homebrewery#198]
**Browser Type/Version**: n/a

**Issue Description**:
Dockerfile does not create working container.

**Code to reproduce**: 
Clone the repo, run `docker build -t naturalcrit .`

**Further information**
Final lines output:

```
npm WARN homebrewery@2.3.0 No repository field.
npm info ok 
 ---> b15f09ed264f
Removing intermediate container e83050364435
Step 17 : RUN gulp fresh
 ---> Running in 8249b781229a
[20:08:18] Local gulp not found in /opt/apps/naturalcrit
[20:08:18] Try running: npm install gulp
```

Same thing on both Arch and Ubuntu 14.04, both with Docker 1.12.1.

it appears that `npm` and `gulp` installations are no longer persisted between intermediate containers. The following changes allow container creation to complete, but `start.sh` fails on a `docker run`:

``` docker
FROM node:latest

MAINTAINER David Hudson <jendave@yahoo.com>

# System update
RUN apt-get -q -y update && \
    apt-get -q -y install npm mongodb --no-install-recommends && \
    apt-get clean && rm -r /var/lib/apt/lists/*

EXPOSE 22 
EXPOSE 8000

ADD start.sh /start.sh 
RUN chmod +x /start.sh 

VOLUME ["/opt/apps"]
COPY . /opt/apps/naturalcrit/
WORKDIR /opt/apps/naturalcrit/

RUN mkdir -p /opt/apps/naturalcrit/node_modules && \
    npm install && \
    npm install -g gulp && \
    npm install gulp && \
    gulp fresh

CMD ["/start.sh"] 
```

**Comment by jonathonf** [comment]
Here you go: https://github.com/stolksdorf/homebrewery/pull/200

This also switches to Alpine and reduces the image size to ~260MB.

**Comment by jonathonf** [comment]
Just as reference, I think the issues I have with the current Dockerfile are due to the use of a volume to contain the application code rather than just data. (This is fixed in the PR.)

**Comment by stolksdorf** [comment]
I've never used docker with this project, so I'm glad @jonathonf is here to save the day. I'll close this once the PR is in.

**Comment by jonathonf** [comment]
Righty, now #203 is merged I've re-created the PR as #206.

**Comment by calculuschild** [comment]
Docker now has updated build files with documentation and should be working.
